lib/Conversion/StandardToHandshake/StandardToHandshake.cpp

```diff
@@ -1660,6 +1660,78 @@ struct HandshakeAnalysisPass
 };
 } // namespace
 
+// This pass iteratively checks the use count of each value, and reformats the
+// MLIR such that every value has exact one use. If a value has multiple uses, a
+// fork is inserted. If a value has no use, a sink is inserted.
+void insertOpForExactOneUse(Value value, handshake::FuncOp funcOp,
+                            OpBuilder &rewriter) {
+
+  if (value.use_empty())
+    rewriter.create<SinkOp>(funcOp.getLoc(), value);
+  else {
+    auto uses = value.getUses();
+    auto outs = std::distance(cbegin(uses), cend(uses));
+    std::vector<Operation *> ops;
+    std::vector<unsigned> idx;
+    for (auto &use : uses) {
+      ops.push_back(use.getOwner());
+      idx.push_back(use.getOperandNumber());
+    }
+
+    // There is an unkown issue when forking the function arguments directly
```

Comment:
I would look into this further, note where the bug is and file a bug report either in the MLIR bug tracker or ours. Please include more information here than 'unknown'. A link to a bug report would be ideal.